desktopexporter/app/routes/main-view.tsx

```diff
@@ -27,50 +47,103 @@ export default function MainView() {
           isFullWidth={isFullWidth}
           toggleSidebarWidth={setFullWidth.toggle}
           traceSummaries={[]}
+          numNewTraces={0}
         />
         <EmptyStateView />
       </Flex>
     );
   }
 
-  let sidebarSummaries: TraceSummaryWithUIData[] =
-    getTraceSummariesWithUIData(traceSummaries);
   return (
     <Flex height="100vh">
       <Sidebar
         isFullWidth={isFullWidth}
         toggleSidebarWidth={setFullWidth.toggle}
-        traceSummaries={sidebarSummaries}
+        traceSummaries={sidebarData.summaries}
+        numNewTraces={sidebarData.numNewTraces}
       />
       <Outlet />
     </Flex>
   );
 }
 
-function getTraceSummariesWithUIData(
+function initSidebarData(traceSummaries: TraceSummary[]): SidebarData {
+  return {
+    summaries: traceSummaries.map((traceSummary) =>
+      generateTraceSummaryWithUIData(traceSummary),
+    ),
+    numNewTraces: 0,
+  };
+}
+
+function updateSidebarData(
+  sidebarData: SidebarData,
   traceSummaries: TraceSummary[],
-): TraceSummaryWithUIData[] {
-  return traceSummaries.map((traceSummary) => {
-    if (traceSummary.hasRootSpan) {
-      let duration = getDurationNs(
-        traceSummary.rootStartTime,
-        traceSummary.rootEndTime,
-      );
-
-      let durationString = getDurationString(duration);
-      return {
-        hasRootSpan: true,
-        rootServiceName: traceSummary.rootServiceName,
-        rootName: traceSummary.rootName,
-        rootDurationString: durationString,
-        spanCount: traceSummary.spanCount,
-        traceID: traceSummary.traceID,
-      };
+): SidebarData {
+  
```

Comment:
nit: this extra space is different from the way you wrote the other functions

Reply:
All gone! 😁